tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/binaryoperations/BinaryExprTest.java

```diff
@@ -143,6 +143,17 @@ public void bitwiseAndTest() {
         Assert.assertEquals(((Byte) returns.get(3)).longValue(), b & d);
     }
 
+    @Test(description = "Test binary expression with query")
+    public void binaryExpressionWithQuery() {
+        Object resultAnd = BRunUtil.invoke(result, "binaryANDWithQuery", new Object[]{});
+        Assert.assertSame(resultAnd.getClass(), Boolean.class);
+        Assert.assertTrue((boolean) resultAnd);
+
+        Object resultOr = BRunUtil.invoke(result, "binaryORWithQuery", new Object[]{});
+        Assert.assertSame(resultOr.getClass(), Boolean.class);
+        Assert.assertTrue((boolean) resultOr);
```

Comment:
The assertions can be done in Ballerina itself.

Also use a data provider or different tests so that the second one will run even if the first one fails.